### PR TITLE
Paratest

### PR DESCRIPTION
**Pull request by chrfritsch**

### CODE REVIEW

File: lib/stages/run_tests.sh [hunk 1]
```diff
@@ -33,7 +45,7 @@ _stage_run_tests() {
         test_selection="${test_selection} --filter ${DRUPAL_TESTING_TEST_FILTER}"
     fi
 
-    local runtest="php ${phpunit} --verbose --debug --configuration ${docroot}/core ${test_selection} ${project_location}"
+    local runtest="php ${phpunit} --verbose --configuration ${docroot}/core ${test_selection} ${project_location}"
```
Comment (chrfritsch) [inline lib/stages/run_tests.sh:48]: Is --debug not available for paratest?

File: lib/stages/run_tests.sh [hunk 2]
```diff
@@ -11,7 +11,19 @@ _stage_run_tests() {
...
+    if ${DRUPAL_TESTING_PARALLEL_TESTING}; then
+        phpunit=${DRUPAL_TESTING_DRUPAL_INSTALLATION_DIRECTORY}/${composer_bin_dir}/paratest" -p "${DRUPAL_TESTING_PARALLEL_TESTING_PROCESSES}
+        if ${DRUPAL_TESTING_PARALLEL_TESTING_PER_FUNCTION}; then
+          phpunit=${phpunit}" -f"
+        fi
+        if ${DRUPAL_TESTING_PARALLEL_TESTING_WRAPPER_RUNNER}; then
+          phpunit=${phpunit}" --runner WrapperRunner"
+        fi
+    else
+        phpunit=${DRUPAL_TESTING_DRUPAL_INSTALLATION_DIRECTORY}/${composer_bin_dir}/phpunit" --debug"
```
Comment (chrfritsch) [inline lib/stages/run_tests.sh:25]: Why this "else" clause? Setting phpunit as a default in line 14 is fine I think

File: configuration.sh [hunk 3]
```diff
@@ -169,6 +169,18 @@ export COMPOSER_MEMORY_LIMIT=${COMPOSER_MEMORY_LIMIT:-"-1"}
 # Memory limit for phpstan.
 export PHPSTAN_MEMORY_LIMIT=${PHPSTAN_MEMORY_LIMIT:-"300M"}
 
+# Enable parallel testing with paratest.
+export DRUPAL_TESTING_PARALLEL_TESTING=${DRUPAL_TESTING_PARALLEL_TESTING:-false}
+
+# Number of parallel processes for paratest.
+export DRUPAL_TESTING_PARALLEL_TESTING_PROCESSES=${DRUPAL_TESTING_PARALLEL_TESTING_PROCESSES:-2}
```
Comment (chrfritsch) [inline configuration.sh:176]: Do we want to set a default? I think paratest "auto" should be the default

File: lib/stages/run_tests.sh [hunk 4]
```diff
@@ -11,7 +11,19 @@ _stage_run_tests() {
     project_location=$(get_project_location)
 
     local test_selection=""
-    local phpunit=${DRUPAL_TESTING_DRUPAL_INSTALLATION_DIRECTORY}/${composer_bin_dir}/phpunit
+    local phpunit=""
+
+    if ${DRUPAL_TESTING_PARALLEL_TESTING}; then
```
Comment (chrfritsch) [inline lib/stages/run_tests.sh:16]: I think if you do a set env in GH actions for this var, it evaluates to a string and then "false" is also true here

**Review by dbosen — APPROVED** [review]

**Pull request merged**